RTech.CoffeeMachine.Api/Program.cs

```diff
@@ -1,10 +1,25 @@
-﻿using RTech.CoffeeMachine.Api.BrewCoffee.Util;
+﻿using RTech.CoffeeMachine.Api.BrewCoffee.Filters;
+using RTech.CoffeeMachine.Api.BrewCoffee.Util;
 
 var builder = WebApplication.CreateBuilder(args);
 var services = builder.Services;
 services.AddMemoryCache();
 services.AddSingleton<IBrewStatusProvider, BrewStatusProvider>();
 services.AddSingleton<IDateTimeProvider, DateTimeProvider>();
+services.AddSingleton<TeapotFilter>();
+services.AddSingleton<BrewStatusUnavailableFilter>();
+
+services.AddHttpClient<IWeatherProxy, WeatherProxy>(client =>
+{
+    var weatherProviderBaseAddress = builder.Configuration.GetValue<string>("WeatherProviderBaseAddress");
+    if (string.IsNullOrWhiteSpace(weatherProviderBaseAddress))
+    {
+        throw new InvalidOperationException("Missing config for WeatherProviderBaseAddress");
+    }
+
+    client.BaseAddress = new Uri(weatherProviderBaseAddress);
+}).SetHandlerLifetime(Timeout.InfiniteTimeSpan);    // Weather API does not require auth (e.g.: no oauth token expiry)
```

Comment:
If OAuth refresh tokens are in play, the refresh handlers would need to be configured at Startup/program.cs & `Timeout.InfiniteTimeSpan` should not be used in that case.